pydruid/query.py

```diff
@@ -266,20 +267,52 @@ def build_query(self, query_type, args):
             elif key == 'datasource':
                 query_dict['dataSource'] = self.parse_datasource(val, query_type)
             elif key == 'paging_spec':
+                # TODO: should paging only exist in outer query?
                 query_dict['pagingSpec'] = val
             elif key == 'limit_spec':
+                # TODO: should limiting only exist in outer query?
                 query_dict['limitSpec'] = val
             elif key == "filter" and val is not None:
                 query_dict[key] = Filter.build_filter(val)
             elif key == "having" and val is not None:
                 query_dict[key] = Having.build_having(val)
             elif key == 'dimension' and val is not None:
-                query_dict[key] = build_dimension(val)
+                if 'inner_dimensions' in args:
+                    query_dict['dimensions'] = [build_dimension(v) for v in args['inner_dimensions']]
```

Comment:
I was a bit confused at first that we handle the "inner_dimensions" both in this query building and in the outer one as well. 
I think that what distinguish nested query from regular query is that the datasource is actually another query, so I feel like if we can keep building the outer query as the regular query and only override the data source with more query that would be easier to read and maintain one flow. WDYT?